python/ql/test/query-tests/Functions/general/functions_test.py

```diff
@@ -198,3 +198,37 @@ def tuple_default(x=(1,2)):
 
 def safe_method(x=[]):
     return x.count(42)
+
+# Modification of parameter with default (sanitised)
+from copy import copy, deepcopy
+def copy_before_modify(x=[]):
+    x = copy(x)
+    return x.append(42)
+
+def deepcopy_before_modify(x=[]):
+    x = deepcopy(x)
+    return x.append(42)
+
+def call_copy_before_modify(x=[]):
+    x = x.copy()
+    return x.append(42)
+
+def unsuccessfully_call_copy_before_modify(x=[]):
+    x = x.reverse().copy() # Unintentially modifies here
```

Comment:
`x.reverse()` reverses `x` in-place and returns `None`, so this code does not work (since `None` has no `copy` method).